MaskRCNN/model/biased_sampler.py

```diff
@@ -9,27 +9,28 @@
 @under_name_scope()
 def sample_fast_rcnn_targets(boxes, gt_boxes, gt_labels, orig_gt_counts, batch_size):
     """
-    Sample some boxes from all proposals for training.
+    Sample boxes according to the predifined fg(foreground) boxes and bg(background) boxes ratio
     #fg(foreground) is guaranteed to be > 0, because ground truth boxes will be added as proposals.
     Args:
-        boxes: (#lvl x BS x K) x 5 region proposals. [batch_index, floatbox] aka Nx5
-        gt_boxes: BS x MaxGT x 4, floatbox
+        boxes: K x 5 region proposals. [batch_index, floatbox]
+        gt_boxes: Groundtruth boxes, BS x MaxGT x 4, floatbox
         gt_labels: BS x MaxGT, int32
-        orig_gt_counts: BS   # The number of ground truths in the data. Use to unpad gt_labels and gt_boxes
+        orig_gt_counts: BS # The number of ground truths in the data. Use to unpad gt_labels and gt_boxes
     Returns:
         sampled_boxes: tx5 floatbox, the rois
         sampled_labels: t int64 labels, in [0, #class). Positive means foreground.
-        fg_inds_wrt_gt: #fg indices, each in range [0, m-1].
-            It contains the matching GT of each foreground roi.
+        fg_inds_wrt_gt: #fg indices, each in range [0, m-1]. It contains the matching GT of each foreground roi.
     """
 
-    per_image_ious = pairwise_iou_batch(boxes, gt_boxes, orig_gt_counts, batch_size=batch_size) # list of len BS [N x M]
+    # Computes pairwise intersection-over-union between rpn boxes and gt boxes
+    # per_image_ious: list of len BS [N x M] -- N is Num_rpn_boxes and M is Num_gt_boxes, for one image
```

Comment:
Num_rpn_boxes is the same as num_rois, right?

Reply:
> Num_rpn_boxes is the same as num_rois, right?

This is the number of boxes before sampling. I would say the num_rois is the number of boxes after sampling.